src/core/blueprint/operators/http.rs

```
@@ -63,13 +89,30 @@ pub fn compile_http(
                 .or(config_module.upstream.on_request.clone())
                 .map(|on_request| HttpFilter { on_request });
 
-            let io = if !http.batch_key.is_empty() && http.method == Method::GET {
+            let group_by_clause = !http.batch_key.is_empty()
+                && (http.method == Method::GET || http.method == Method::POST);
+            let io = if group_by_clause {
                 // Find a query parameter that contains a reference to the {{.value}} key
-                let key = http.query.iter().find_map(|q| {
-                    Mustache::parse(&q.value)
-                        .expression_contains("value")
-                        .then(|| q.key.clone())
-                });
+                let key = if http.method == Method::GET {
+                    http.query.iter().find_map(|q| {
+                        Mustache::parse(&q.value)
+                            .expression_contains("value")
+                            .then(|| q.key.clone())
+                    })
+                } else {
+                    // find the key from the body where value is mustache template.
+                    http.body
+                        .as_ref()
+                        .map(|b| extract_expression_keys(b))
+                        .and_then(|keys| {
+                            if keys.len() == 1 {
+                                Some(keys[0].clone())
+                            } else {
+                                None
```

If you can't identify the key, you should throw an error.